src/lib.rs

```diff
@@ -1062,50 +1064,14 @@ impl Node {
 		let con_addr = peer_info.address.clone();
 		let con_cm = Arc::clone(&self.connection_manager);
 
-		let cur_anchor_reserve_sats =
-			total_anchor_channels_reserve_sats(&self.channel_manager, &self.config);
-		let spendable_amount_sats =
-			self.wallet.get_spendable_amount_sats(cur_anchor_reserve_sats).unwrap_or(0);
-
-		// Fail early if we have less than the channel value available.
-		if spendable_amount_sats < channel_amount_sats {
-			log_error!(self.logger,
-				"Unable to create channel due to insufficient funds. Available: {}sats, Required: {}sats",
-				spendable_amount_sats, channel_amount_sats
-			);
-			return Err(Error::InsufficientFunds);
-		}
-
 		// We need to use our main runtime here as a local runtime might not be around to poll
 		// connection futures going forward.
 		self.runtime.block_on(async move {
 			con_cm.connect_peer_if_necessary(con_node_id, con_addr).await
 		})?;
 
-		// Fail if we have less than the channel value + anchor reserve available (if applicable).
-		let init_features = self
-			.peer_manager
-			.peer_by_node_id(&node_id)
-			.ok_or(Error::ConnectionFailed)?
-			.init_features;
-		let required_funds_sats = channel_amount_sats
-			+ self.config.anchor_channels_config.as_ref().map_or(0, |c| {
-				if init_features.requires_anchors_zero_fee_htlc_tx()
-					&& !c.trusted_peers_no_reserve.contains(&node_id)
-				{
-					c.per_channel_reserve_sats
-				} else {
-					0
-				}
-			});
-
-		if spendable_amount_sats < required_funds_sats {
-			log_error!(self.logger,
-				"Unable to create channel due to insufficient funds. Available: {}sats, Required: {}sats",
-				spendable_amount_sats, required_funds_sats
-			);
-			return Err(Error::InsufficientFunds);
-		}
+		// Check funds availability after connection (includes anchor reserve calculation)
```

Comment:
Ah, that it's now done after connection is clear, but why did we switch to it? Previously we'd abort early to avoid the unnecessary delay/connection establishment, if we don't have sufficient funds anyways?